src/Squidex.Infrastructure.MongoGridFs/Assets/MongoGridFsAssetStore.cs

```diff
@@ -141,16 +141,13 @@ public Task DeleteAsync(string id, long version, string suffix)
         {
             try
             {
-                var filter =
-                    Builders<GridFSFileInfo>.Filter.And(
-                        Builders<GridFSFileInfo>.Filter.Eq(x => x.Filename, file.Name)
-                    );
-                using (var cursor = await bucket.FindAsync(filter))
+                file.Delete();
+                using (var cursor = await bucket.FindAsync(Builders<GridFSFileInfo>.Filter.And(
+                    Builders<GridFSFileInfo>.Filter.Eq(x => x.Filename, file.Name)
```

Comment:
I still do not see why you have to loop. Can you not use use: `bucket.DeleteAsync(file.Name)`?

Reply:
Unfortunately no support from IGridFSBucket to delete fine by name currently.

Reply:
Is the name not the same like Id?

Reply:
Here it looks like it could be anything: https://mongodb.github.io/mongo-csharp-driver/2.4/apidocs/html/T_MongoDB_Driver_GridFS_GridFSBucket_1.htm

Reply:
nope, it is not

Reply:
it is object identifier "Id" from GridFSFileInfo 
http://api.mongodb.com/csharp/current/html/T_MongoDB_Driver_GridFS_GridFSFileInfo.htm

Reply:
It says that there is a generic alternative, so you could use IGridFSBucket<string> and use FileName (Name+version+suffix) as Id.

Reply:
yes

Reply:
it will also overcomplicate bucket.UploadFromStreamAsync(file.Name, stream, cancellationToken: ct);

Reply:
Perhaps I have another version, but for me it looks like

`bucket.UploadFromStreamAsync(file.Name, file.Name, stream, ...);`

https://mongodb.github.io/mongo-csharp-driver/2.4/apidocs/html/M_MongoDB_Driver_GridFS_GridFSBucket_1_UploadFromStreamAsync.htm